src/tokenizer.rs

```diff
@@ -1091,6 +1122,41 @@ mod tests {
         compare(expected, tokens);
     }
 
+    #[test]
+    fn tokenize_select_exponent() {
+        let sql = String::from("SELECT 1e10, 1e-10, 1e+10, 1ea, 1e-10a, 1e-10-10");
+        let dialect = GenericDialect {};
+        let mut tokenizer = Tokenizer::new(&dialect, &sql);
+        let tokens = tokenizer.tokenize().unwrap();
+
+        let expected = vec![
+            Token::make_keyword("SELECT"),
+            Token::Whitespace(Whitespace::Space),
+            Token::Number(String::from("1e10"), false),
+            Token::Comma,
+            Token::Whitespace(Whitespace::Space),
+            Token::Number(String::from("1e-10"), false),
+            Token::Comma,
+            Token::Whitespace(Whitespace::Space),
+            Token::Number(String::from("1e+10"), false),
+            Token::Comma,
+            Token::Whitespace(Whitespace::Space),
+            Token::Number(String::from("1"), false),
+            Token::make_word("ea", None),
+            Token::Comma,
+            Token::Whitespace(Whitespace::Space),
+            Token::Number(String::from("1e-10"), false),
+            Token::make_word("a", None),
```

Comment:
I found this very strange that a new token is formed without whitespace after a number. I expected that this is a token error but this implementation agrees with postgres 🤯 

```sql
postgres=# select 12e-10a;
      a       
--------------
 0.0000000012
(1 row)

postgres=# select 12e-10 a;
      a       
--------------
 0.0000000012
(1 row)
```

Reply:
Likewise 

```sql
postgres=# select 1e-10-10;
   ?column?    
---------------
 -9.9999999999
(1 row)

postgres=# select 1e-10 -10;
   ?column?    
---------------
 -9.9999999999
(1 row)
```

🤯

Reply:
I believe this behaviour is part of what bit me when trying to implement for Hive dialect 😅